openverse_catalog/dags/providers/provider_api_scripts/science_museum.py

```diff
@@ -173,60 +173,50 @@ def _get_creator_info(obj_attr):
 
 
-def _get_filesize(image_url):
-    """
-    Get the size of the image in bytes.
-    """
-    resp = delayed_requester.get(image_url)
-    if resp:
-        filesize = int(resp.headers.get("Content-Length", 0))
-        return filesize if filesize != 0 else None
-
-
-def _get_image_info(processed):
+def _get_image_info(
+    processed: Dict,
+) -> Tuple[str | None, int | None, int | None, str | None]:
+    height, width, filetype = None, None, None
     image_data = processed.get("large")
     if image_data is None:
         image_data = processed.get("medium", {})
 
-    image = image_data.get("location")
-    measurements = image_data.get("measurements")
-    filetype = image_data.get("format")
-    image = check_url(image)
-    height, width = _get_dimensions(measurements)
-    filesize = _get_filesize(image)
-    return image, height, width, filetype, filesize
+    image_url = check_url(image_data.get("location"))
+    if image_url:
+        filetype = image_data.get("format")
+        height, width = _get_dimensions(image_data)
+    return image_url, height, width, filetype
 
 
-def check_url(image_url):
-    base_url = "https://coimages.sciencemuseumgroup.org.uk/images/"
-    if image_url:
-        if "http" in image_url:
-            checked_url = image_url
-        else:
-            checked_url = base_url + image_url
-    else:
-        checked_url = None
-    return checked_url
+def check_url(image_url: str | None) -> str | None:
+    if not image_url:
+        return None
+    if image_url.startswith("http"):
+        return image_url
+    return f"https://coimages.sciencemuseumgroup.org.uk/images/{image_url}"
 
 
-def _get_dimensions(measurements):
-    height_width = {}
-    if measurements:
-        dimensions = measurements.get("dimensions")
-        if dimensions:
-            for dim in dimensions:
-                height_width[dim.get("dimension")] = dim.get("value")
-    return height_width.get("height"), height_width.get("width")
+def _get_dimensions(image_data: Dict) -> Tuple[int | None, int | None]:
+    """
+    Returns the height and width of the image from a list of dictionaries
+    with keys of "dimension", "units", "value".
+    """
+    size = {}
+    dimensions = image_data.get("measurements", {}).get("dimensions")
+    if dimensions:
+        for dim in dimensions:
+            size[dim.get("dimension")] = (
+                dim.get("value") if dim.get("units") == "pixels" else None
+            )
+    return size.get("height"), size.get("width")
 
 
-def _get_license_version(source):
+def _get_license_version(source: Dict | None) -> str | None:
     license_version = None
     if source:
-        legal = source.get("legal")
-        if legal:
-            rights = legal.get("rights")
-            if type(rights) == list:
-                license_version = rights[0].get("usage_terms")
+        rights = source.get("legal", {}).get("rights")
+        if isinstance(rights, list) and len(rights) > 0:
```

Comment:
Nice simplification! Maybe this could even be?
```suggestion
        rights = source.get("legal", {}).get("rights", [])
        if rights:
```

This defaults to a list and just does a truthy check on the list to see if it has items. I guess the `isinstance` check would be important if `rights` could ever be an truthy object that's _not_ a list, like a dict or something 🤔